app/controllers/concerns/authentication_concerns.rb

```diff
@@ -3,14 +3,26 @@ module AuthenticationConcerns
 
   included do
     helper_method :authenticated?
+    helper_method :current_organisation
     helper_method :current_school
+    helper_method :current_school_group
   end
 
   def authenticated?
     session[:session_id].present?
   end
 
+  def current_organisation
+    current_school || current_school_group
+  end
+
   def current_school
-    School.find_by!(urn: session[:urn])
+    @current_school ||= School.find_by!(urn: session[:urn]) if session[:urn].present?
+  end
+
+  def current_school_group
+    if SchoolGroupJobsFeature.enabled?
+      @current_school_group ||= SchoolGroup.find_by!(uid: session[:uid]) if session[:uid].present?
```

Comment:
Is it possible for these instance variables to outlive the deletion of the session?

Reply:
If they could, that would be no different to what is currently the case. But if the session is deleted then you're not signed in